### ISSUE TITLE
[Accepted] SDL 0226 - Add iOS SDLManager RPC Subscriptions

### ISSUE THREAD

**Issue by theresalech** [smartdevicelink/sdl_evolution#714]
Hello SDL community,

The review of "SDL 0226 - Add iOS SDLManager RPC Subscriptions" begins now and runs through May 7, 2019.  The proposal is available here:

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0226-ios-manager-rpc-subscriptions.md

Reviews are an important part of the SDL evolution process. All reviews should be sent to the associated Github issue at:

https://github.com/smartdevicelink/sdl_evolution/issues/714

What goes into a review?

The goal of the review process is to improve the proposal under review through constructive criticism and, eventually, determine the direction of SDL. When writing your review, here are some questions you might want to answer in your review:

* Is the problem being addressed significant enough to warrant a change to SDL?
* Does this proposal fit well with the feel and direction of SDL?
* If you have used competitors with a similar feature, how do you feel that this proposal compares to those?
* How much effort did you put into your review? A glance, a quick reading, or an in-depth study?
Please state explicitly whether you believe that the proposal should be accepted into SDL.

More information about the SDL evolution process is available at

https://github.com/smartdevicelink/sdl_evolution/blob/master/process.md

Thank you,
Theresa Lech

Program Manager - Livio
theresa@livio.io

**Comment by joeygrover** [comment]
The only issue I have with this proposal is Android currently has separate listeners for each type of RPC with specific interfaces. I would like to see the two platforms align in the API either way.

**Comment by joeljfischer** [comment]
I understand the concern. I originally had it mimicking the Java Suite listener more closely. However, the reason I chose to go in a different direction was:

The iOS library currently has a way to subscribe to RPC callbacks through the iOS `NotificationCenter` API and all of the notification names are already public APIs. As a result, I didn't want to create a second, entirely separate way to do it through the `SDLManager`. Instead, I decided to keep using the existing public API and simply expose an API that uses the `NotificationCenter` API under the hood. This keeps the iOS implementation consistent with the existing API (and only a slight departure from Java's).

Alternatively, we could deprecate all of the existing iOS RPC callback strings and re-do it to be closer to the Android API (by passing the RPC name and type).
